app/src/main/java/com/zulip/android/filters/NarrowFilterByDate.java

```diff
@@ -35,44 +37,45 @@ public NarrowFilterByDate(Date date) {
         this.date = date;
     }
 
-    /**
-     * Checks two dates are of same day or not
-     *
-     * @param date1 long date1 to be compared with date2
-     * @param date2 long date2 to be compared with date1
-     * @return boolean
-     */
-    private static boolean isSameDay(long date1, long date2) {
-        return date1 / 86400000 == date2 / 86400000;
-    }
-
     public int describeContents() {
         return 0;
     }
 
     public void writeToParcel(Parcel dest, int flags) {
+        dest.writeValue(date);
```

Comment:
When you write data to parcel, it is analogous to serialization, so we also need to deserialize the result to get back the NarrowFilterByDate instance in [creator createfromparcel](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/filters/NarrowFilterByDate.java#L19).